crates/router/src/connector/globalpay/transformers.rs

```diff
@@ -341,3 +327,127 @@ pub struct GlobalpayErrorResponse {
     pub detailed_error_code: String,
     pub detailed_error_description: String,
 }
+
+fn get_payment_method_data(
+    item: &types::PaymentsAuthorizeRouterData,
+    brand_reference: Option<String>,
+) -> Result<PaymentMethodData, error_stack::Report<errors::ConnectorError>> {
+    match item.request.payment_method_data.clone() {
+        api::PaymentMethodData::Card(ccard) => Ok(PaymentMethodData::Card(requests::Card {
+            number: ccard.card_number,
+            expiry_month: ccard.card_exp_month,
+            expiry_year: ccard.card_exp_year,
+            cvv: ccard.card_cvc,
+            account_type: None,
+            authcode: None,
+            avs_address: None,
+            avs_postal_code: None,
+            brand_reference,
+            chip_condition: None,
+            funding: None,
+            pin_block: None,
+            tag: None,
+            track: None,
+        })),
+        api::PaymentMethodData::Wallet(wallet_data) => get_wallet_data(&wallet_data),
+        api::PaymentMethodData::BankRedirect(bank_redirect) => {
+            get_bank_redirect_data(&bank_redirect)
+        }
+        _ => Err(errors::ConnectorError::NotImplemented(
+            "Payment methods".to_string(),
+        ))?,
+    }
+}
+
+fn get_return_url(item: &types::PaymentsAuthorizeRouterData) -> Option<String> {
+    match item.request.payment_method_data.clone() {
+        api::PaymentMethodData::Wallet(api_models::payments::WalletData::PaypalRedirect(_)) => {
+            item.request.complete_authorize_url.clone()
+        }
+        _ => item.request.router_return_url.clone(),
+    }
+}
+
+type MandateDetails = (Option<Initiator>, Option<StoredCredential>, Option<String>);
+fn get_mandate_details(
+    item: &types::PaymentsAuthorizeRouterData,
+) -> Result<MandateDetails, error_stack::Report<errors::ConnectorError>> {
+    let connector_mandate_id = item
+        .request
+        .mandate_id
+        .as_ref()
+        .and_then(|mandate_ids| mandate_ids.connector_mandate_id.clone());
+    Ok(if is_mandate_payment(item, connector_mandate_id.as_ref()) {
+        (
+            Some(match item.request.off_session {
+                Some(true) => Initiator::Merchant,
+                _ => Initiator::Payer,
+            }),
+            Some(StoredCredential {
+                model: Some(requests::Model::Recurring),
+                sequence: Some(match connector_mandate_id.is_some() {
+                    true => requests::Sequence::Subsequent,
+                    false => requests::Sequence::First,
+                }),
+            }),
+            connector_mandate_id,
+        )
+    } else {
+        (None, None, None)
+    })
+}
+
+fn is_mandate_payment(
+    item: &types::PaymentsAuthorizeRouterData,
+    connector_mandate_id: Option<&String>,
+) -> bool {
+    item.request.setup_mandate_details.is_some() || connector_mandate_id.is_some()
+}
```

Comment:
```suggestion
fn is_mandate_payment(item: &types::PaymentsAuthorizeRouterData) -> bool {
    item.request.setup_mandate_details.is_some()
        || item
            .request
            .mandate_id
            .as_ref()
            .and_then(|mandate_ids| mandate_ids.connector_mandate_id.as_ref())
            .is_some()
}
```
PaymentsAuthorizeRouterData alone is sufficient to be passed to the method. 
We can move this into utils so that this code can be reused

Reply:
I have moved this to connector utils